wagtail/models/__init__.py

```diff
@@ -2172,21 +2216,41 @@ class PageRevision(models.Model):
     )
 
     objects = models.Manager()
+    page_revisions = PageRevisionsManager()
     submitted_revisions = SubmittedRevisionsManager()
 
+    @cached_property
+    def content_object(self):
+        return self.base_content_type.get_object_for_this_type(pk=self.object_id)
+
+    @cached_property
+    def page(self):
+        return self.content_object
+
+    @cached_property
+    def page_id(self):
+        return int(self.object_id)
```

Comment:
We should either get rid of this now or add some documentation and deprecation warning and get rid of this later. Or maybe leave this in as a shorthand, but make it undocumented?

Reply:
I haven't got a strong opinion on this. @gasman what do you think?

Reply:
I'd go for the deprecation. It seems like something people are likely to be using in the wild, but that we don't want to keep around indefinitely.

Reply:
Should this be removed in Wagtail 5.0, or 6.0?

Reply:
Let's say 5.0, since we don't know at this point how far into the future 5.0 and 6.0 are going to be. (I know this tripped us up with `use_json_field`, and we're still figuring things out - but I think we can make it work as long as we hold off on doing the actual removals for version N until version N-1 has been released and bakerydemo is updated accordingly).

Reply:
Okay, I've added the deprecation warnings. Let me know if there's any changes needed.